anomaly_agent/models.py

```
@@ -0,0 +1,65 @@
+            except ValueError:
+                try:
+                    dt = datetime.strptime(v, "%Y-%m-%d")
+                    return dt.strftime("%Y-%m-%d %H:%M:%S")
+                except ValueError:
+                    try:
+                        datetime.strptime(v, "%Y-%m-%d %H:%M:%S")
+                        return v
+                    except ValueError:
+                        raise ValueError(
+                            f"timestamp must be in {TIMESTAMP_FORMAT} format, ISO format, or YYYY-MM-DD format"
+                        )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)